src/binfhe/examples/pke/boolean-ap-pke.cpp

```diff
@@ -0,0 +1,97 @@
+//==================================================================================
+// BSD 2-Clause License
+//
+// Copyright (c) 2014-2022, NJIT, Duality Technologies Inc. and other contributors
+//
+// All rights reserved.
+//
+// Author TPOC: contact@openfhe.org
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+// 1. Redistributions of source code must retain the above copyright notice, this
+//    list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright notice,
+//    this list of conditions and the following disclaimer in the documentation
+//    and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//==================================================================================
+
+/*
+  Example for the FHEW scheme using the AP bootstrapping
+ */
+
+#include "binfhecontext.h"
+
+using namespace lbcrypto;
+
+int main() {
+    // Sample Program: Step 1: Set CryptoContext
+
+    auto cc = BinFHEContext();
+
+    // STD128 is the security level of 128 bits of security based on LWE Estimator
+    // and HE standard. Other common options are TOY, MEDIUM, STD192, and STD256.
+    // MEDIUM corresponds to the level of more than 100 bits for both quantum and
+    // classical computer attacks. The second argument is the bootstrapping method
+    // (AP or GINX). The default method is GINX. Here we explicitly set AP. GINX
+    // typically provides better performance: the bootstrapping key is much
+    // smaller in GINX (by 20x) while the runtime is roughly the same.
+    cc.GenerateBinFHEContext(STD128, AP);
+
+    // Sample Program: Step 2: Key Generation
+
+    // Generate the secret key
+    auto sk = cc.KeyGen();
+
+    std::cout << "Generating the bootstrapping keys..." << std::endl;
+
+    // Generate the bootstrapping keys (refresh, switching and public keys)
+    // Public keys are generated when the publicKeyFlag is set to true
+    cc.BTKeyGen(sk, true);
```

Comment:
I think using an enum would be better. It could be SKONLY, KEYPAIR. It might be extended to other options later, e.g., when we add threshold FHE or multi-key FHE in the future.